### PR TITLE
refactor: consolidate metrics-server addon spec

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR consolidates the maintenance of the `metrics-server` addon into a single go template-aware YAML file.

**Issue Fixed**:

Addresses #3188 

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/addons.go [hunk 1]
```diff
@@ -254,6 +255,10 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		},
 	}
 
+	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.16.0") {
+		defaultMetricsServerAddonsConfig.Mode = AddonModeReconcile
```
Comment (mboersma) [inline pkg/api/addons.go:259]: Was there a good reason for `EnsureExists` here before 1.16? It's odd that they differed.

Also if `Reconcile` is the default from 1.16 onward, maybe it should be the default and this block could set `EnsureExists` for earlier versions. Then when 1.16 support drops off, this block could be deleted altogether.

File: pkg/engine/const.go [hunk 2]
```diff
@@ -199,8 +199,8 @@ const (
 const (
 	heapsterAddonSourceFilename                    string = "kubernetesmasteraddons-heapster-deployment.yaml"
 	heapsterAddonDestinationFilename               string = "kube-heapster-deployment.yaml"
-	metricsServerAddonSourceFilename               string = "kubernetesmasteraddons-metrics-server-deployment.yaml"
-	metricsServerAddonDestinationFilename          string = "kube-metrics-server-deployment.yaml"
+	metricsServerAddonSourceFilename               string = "metrics-server.yaml"
```
Comment (mboersma) [inline pkg/engine/const.go:202]: At some point maybe we should fold source and destination into just a single `metricsServerAddonFilename`, for example.

Comment (jackfrancis) [reply pkg/engine/const.go:202]: yeah, that would probably be easier, just not changing that yet becasue reasons

File: pkg/api/addons.go (hunk 1 above)
Comment (jackfrancis) [reply pkg/api/addons.go:259]: I can imagine that we decided to prefer `Reconcile` at a time when we were introducing 1.15, and didn't want to change pre-existing versions?

I like your idea.

**Pull request merged**